Packs/VeloCloud/Integrations/VeloCloudEventCollector/VeloCloudEventCollector.py

```diff
@@ -0,0 +1,437 @@
+        limit: Maximum number of events to retrieve per page
+
+    Returns:
+        A tuple containing the list of new events and the new last timestamp
+
+    Note:
+        The function increments the start time by 1 second to minimize overlap.
+    """
+    start_time_str, end_time_str = velocloud_parse_date_range(first_fetch, last_timestamp)
+    new_last_timestamp = end_time_str  # If no new events, set to end_time to avoid searching same range again
+
+    demisto.debug(f"Calling client.get_events with start_time: {start_time_str} and end_time: {end_time_str}")
+
+    response = client.get_events(start_time=start_time_str, end_time=end_time_str, limit=limit)
+
+    events = format_events(response)
+
+    demisto.debug(f"Events: {json.dumps(events)}")
+
+    # Update last timestamp if events exist
+    if events:
+        # Filter out None values and get valid timestamps
+        valid_timestamps = [ts for ts in [event.get("_time") for event in events] if ts and isinstance(ts, str)]
+        if valid_timestamps:
+            new_last_timestamp = max(valid_timestamps)
+
+    demisto.debug(f"New last timestamp: {new_last_timestamp}")
+    return events, new_last_timestamp
+
+
+def fetch_events_command(client: Client) -> None:
+    """Fetch events and send them to XSIAM
+
+    Args:
+        client: VeloCloud client
+    """
+    params = demisto.params()
+    first_fetch = params.get("first_fetch", "1 day ago")
+    limit = arg_to_number(params.get("max_fetch", 1000))
+
+    last_run = demisto.getLastRun()
+    last_timestamp = last_run.get("last_timestamp", "")
+
+    if not last_timestamp:
+        start_time, _ = velocloud_parse_date_range(first_fetch, "")
+        last_timestamp = start_time
+
+    demisto.debug(f"Last run dict: {last_run}")
+    demisto.debug(f"Calling get_events with last_timestamp: {last_timestamp} and first_fetch: {first_fetch}")
+
+    events, new_last_timestamp = fetch_events(
+        client=client,
+        last_timestamp=last_timestamp,
+        first_fetch=first_fetch,
+        limit=limit if limit is not None else 1000,
+    )
+
+    demisto.debug(f"Sending {len(events)} events to send_events_to_xsiam")
+    send_events_to_xsiam(events, VENDOR, PRODUCT)
+
+    new_last_run = {"last_timestamp": new_last_timestamp}
+    demisto.debug(f"New last run: {json.dumps(new_last_run)}")
+    demisto.setLastRun(new_last_run)
+
+
+def main():
+    """Main function, parses params and runs command functions"""
+    params = demisto.params()
+
+    # Get connection parameters
+    base_url = params.get("url", "")
+    enterprise_id = params.get("enterprise_id", "")
+    api_key = params.get("api_key", {}).get("password", "")
+    api_key = api_key.strip() if api_key else ""
+
+    # Check required parameters based on auth type
+    if not api_key:
+        raise ValueError("API Key is required for authentication")
+
+    if not enterprise_id:
+        raise ValueError("Enterprise ID is required")
```

Comment:
remove these lines and set the params as required in the yml, in that case the instance wont be saved unless they're defined
```suggestion
```